wemake_python_styleguide/visitors/ast/complexity/counts.py

```diff
@@ -81,11 +82,7 @@ def _post_visit(self) -> None:
             )
 
 
-@final
-@alias('visit_any_import', (
-    'visit_ImportFrom',
-    'visit_Import',
-))
+@final  # noqa: WPS214
```

Comment:
Why do we have a violation here? Can we remove it?

Reply:
there is too many methods in class, because I had to use 2 methods specific to each import node (now `ImportFrom` node has one check more than `Import`).
I can remove it by:

+ using old implementation and call this check only if `isinstance(node, ast.ImportFrom)`
+ move check implementation from separate method to `.visit_ImportFrom()`
+ extract some methods to module level functions/classes

What do you think?

Reply:
I always prefer `extract some methods to module level functions/classes` if you don't mind.

Reply:
Generally I am also fan of this approach, but in this case every method either use some class' instance attributes (`.options`) or update instance's attributes (`._imports_count` and `_imported_names_count`) and these methods don't have much in common (maybe except `._check_import_count()` and `._check_imported_names_count()`), so I don't have good idea which methods should be extracted.
Maybe I should create `ImportFromMembersNumberValidator` similar to `ImportFromValidator` from `imports` visitor and just use it in `visit_ImportFrom()`, what do you think?

Reply:
Whatever you think is best! 👍

Reply:
Ok, so now, when I have extracted some logic to new class, another problem appeared - there is to many module members. Can I ignore this violation or should we split `wemake_python_styleguide/visitors/ast/complexity/counts.py` into multiple files?

Reply:
No, ignoring complexity violations is not an option, sorry: https://sobolevn.me/2019/10/complexity-waterfall

We can split this module:
1. method related counts can be moved to `classes.py`
2. logic can be moved to `logic/`